client/src/layers/TextLayer.ts

```diff
@@ -11,32 +11,70 @@ export interface TextData {
   y: number;
   offsetY?: number;
   offsetX?: number;
+  currentPair?: boolean;
 }
 
 export type FormatTextRow = (track: FrameDataTrack) => TextData[] | null;
 
 interface TextLayerParams {
   formatter?: FormatTextRow;
 }
-
 /**
+ * @param track - standard frameDataTrack info
+ * @param additionalNum Number of additional pairs to display, top pair is the current pair
+ * followed by more in descending order.
  * @returns value or null.  null indicates that the text should not be displayed.
  */
-function defaultFormatter(track: FrameDataTrack): TextData[] | null {
+function defaultFormatter(track: FrameDataTrack, additionalNum = 0): TextData[] | null {
   if (track.features && track.features.bounds) {
     const { bounds } = track.features;
+
     if (bounds && track.confidencePairs !== null) {
-      const type = track.confidencePairs[0];
-      const confidence = track.confidencePairs[1];
-      return [{
+      const lineHeight = 25;
+      const maxPairs = Math.min(track.confidencePairs.length, additionalNum);
+      let currentHeight = bounds[1] - (lineHeight * (maxPairs + 1));
+      const arr: TextData[] = [];
+      const baseType = track.trackType ? track.trackType[0] : 'unknown';
+      const baseConfidence = track.trackType ? track.trackType[1] : 1.0;
```

Comment:
I don't think this is a good place to do defaults.

Tracks without confidence pairs shouldn't be allowed at all into this application.  `unknown: 1.0` defaults should be forced at ingest or, really, by the backend.  I actually think the frontend should refuse to load at all and show an error message, because a track without confidence pairs is malformed data.

This seems out-of-scope for this PR.